server/sonar-server/src/main/java/org/sonar/server/branch/pr/ws/ListAction.java

```diff
@@ -0,0 +1,142 @@
+      .setSince("7.1")
+      .setDescription("List the pull requests of a project.<br/>" +
+        "Requires 'Administer' rights on the specified project.")
+      .setResponseExample(getClass().getResource("list-example.json"))
+      .setHandler(this);
+
+    addProjectParam(action);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    String projectKey = request.mandatoryParam(PARAM_PROJECT);
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      ComponentDto project = componentFinder.getByKey(dbSession, projectKey);
+      userSession.checkComponentPermission(UserRole.USER, project);
+      checkArgument(project.isEnabled() && PROJECT.equals(project.qualifier()), "Invalid project key");
+
+      List<BranchDto> pullRequests = dbClient.branchDao().selectByComponent(dbSession, project).stream()
+        .filter(b -> b.getBranchType() == PULL_REQUEST)
+        .collect(toList());
+      List<String> pullRequestUuids = pullRequests.stream().map(BranchDto::getUuid).collect(toList());
+
+      Map<String, BranchDto> mergeBranchesByUuid = dbClient.branchDao()
+        .selectByUuids(dbSession, pullRequests.stream().map(BranchDto::getMergeBranchUuid).filter(Objects::nonNull).collect(toList()))
+        .stream().collect(uniqueIndex(BranchDto::getUuid));
+      Map<String, BranchStatistics> branchStatisticsByBranchUuid = issueIndex.searchBranchStatistics(project.uuid(), pullRequestUuids).stream()
+        .collect(uniqueIndex(BranchStatistics::getBranchUuid, Function.identity()));
+      Map<String, String> analysisDateByBranchUuid = dbClient.snapshotDao().selectLastAnalysesByRootComponentUuids(dbSession, pullRequestUuids).stream()
+        .collect(uniqueIndex(SnapshotDto::getComponentUuid, s -> formatDateTime(s.getCreatedAt())));
+
+      ProjectPullRequests.ListWsResponse.Builder protobufResponse = ProjectPullRequests.ListWsResponse.newBuilder();
+      pullRequests
+        .forEach(b -> addPullRequest(protobufResponse, b, mergeBranchesByUuid, branchStatisticsByBranchUuid.get(b.getUuid()),
+          analysisDateByBranchUuid.get(b.getUuid())));
+      writeProtobuf(protobufResponse.build(), request, response);
+    }
+  }
+
+  private static void addPullRequest(ProjectPullRequests.ListWsResponse.Builder response, BranchDto branch, Map<String, BranchDto> mergeBranchesByUuid,
+    BranchStatistics branchStatistics, @Nullable String analysisDate) {
+    Optional<BranchDto> mergeBranch = Optional.ofNullable(mergeBranchesByUuid.get(branch.getMergeBranchUuid()));
+
+    ProjectPullRequests.PullRequest.Builder builder = ProjectPullRequests.PullRequest.newBuilder();
+    builder.setId(branch.getKey());
+
+    // known to be non-null; all pull requests have non-null value, and caller filters accordingly
+    DbProjectBranches.PullRequestData pullRequestData = branch.getPullRequestData();
+    builder.setBranch(pullRequestData.getBranch());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "pullRequestData" is nullable here. ([squid:S2259](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fvsts_pr&issues=AWHB4367YRImEm_EB1Ud&open=AWHB4367YRImEm_EB1Ud)

[ISSUE_KEY]: AWHB4367YRImEm_EB1Ud